Fix #3474: Merge faq single activity into single layout file.

rishidyno opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation

Fix #3474

<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

rishidyno committed:
in progress

rishidyno committed:
in progress

rishidyno committed:
in progress

rishidyno committed:
in progress

rishidyno committed:
merged faq_single_activity.xml into one file

rishidyno committed:
merged faq_single_activity.xml into one file

rishidyno committed:
Merge remote-tracking branch 'origin/develop' into merge-faq_single_activity-into-single-layout-file

rishidyno commented:
|hd|dds|
|-------|-------|
| ddc| sss|
|ds|fae|

rishidyno commented:
| OLD Screenshots         | NEW Screenshots |
| -----------       | -----------      |
| ![Screenshot_20210727_025302](https://user-images.githubusercontent.com/78549195/127765337-fb392b8d-70e4-4fdc-8b90-0794994cfbc5.png) | ![Screenshot_20210727_024438](https://user-images.githubusercontent.com/78549195/127765341-b24ebdec-f8aa-4035-8899-86a52be35fd1.png) |
| ![Screenshot_20210727_025318](https://user-images.githubusercontent.com/78549195/127765365-3cc8b695-bc38-431f-9bdd-b483477aba9f.png) | ![Screenshot_20210727_024330](https://user-images.githubusercontent.com/78549195/127765369-a4e34547-a9b6-4405-9c37-eca0c2e5d835.png) |
| ![Screenshot_20210727_025340](https://user-images.githubusercontent.com/78549195/127765393-38c0666b-4c30-476e-b6ec-b403714338ab.png) | ![Screenshot_20210727_024527](https://user-images.githubusercontent.com/78549195/127765396-20c33062-53d3-4721-aa7d-0afef60ec671.png) | 
|![Screenshot_20210727_025333](https://user-images.githubusercontent.com/78549195/127765418-829e13b2-f1f0-4efe-9c01-b77af2e4daf9.png) | ![Screenshot_20210804_130410](https://user-images.githubusercontent.com/78549195/128141520-8f4b9b41-6088-480d-b790-71794705e2c5.png) |

rishidyno commented:
@rt4914 I have added the required screenshots and all tests have passed. Please, review my PR

prayutsu commented on app/src/main/res/values-sw600dp-land/booleans.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <bool name="faq_single_activity_scroll_fill_view_port">true</bool>
+</resources>
```
Add EOF

rishidyno replied:
@prayutsu I have added EOF please review it.

prayutsu reviewed (commented):
Generally LGTM, just have one comment.

rishidyno committed:
merged faq_single_activity.xml into one file

rishidyno committed:
added EOL

rishidyno committed:
Added EOL to booleans.xml sw600-dp

rishidyno commented:
@rt4914 @anandwana001 Please review my PR.

prayutsu commented:
@rishidyno PR mostly looks good to me, can you please add the screenshots in a table so that it is easy to compare?

rishidyno committed:
deleted 3 versions of topic_practice_footer_view.xml and merged it into one updated

rishidyno committed:
updated project

anandwana001 commented:
Assign me once all test case gets pass.

rishidyno commented:
Ptal @anandwana001   Assigning this to you.

rishidyno committed:
Merge branch 'oppia:develop' into merge-faq_single_activity-into-single-layout-file

rishidyno committed:
updating project

rishidyno committed:
updating project

rt4914 commented on app/src/main/res/values-land/dimens.xml:
```diff
@@ -229,6 +229,7 @@
   <dimen name="faq_single_padding_top">28dp</dimen>
   <dimen name="faq_single_padding_end">72dp</dimen>
   <dimen name="faq_single_padding_bottom">56dp</dimen>
+  <dimen name="faq_single_activity_scroll_padding_Bottom">108dp</dimen>
```
Rename to `faq_single_activity_scroll_padding_bottom`

rishidyno replied:
Yes Sir I have renamed it to `faq_single_activity_scroll_padding_bottom` in dimens.xml-land

rt4914 commented on app/src/main/res/values-sw600dp-land/dimens.xml:
```diff
@@ -250,6 +250,7 @@
   <dimen name="faq_single_padding_top">48dp</dimen>
   <dimen name="faq_single_padding_end">32dp</dimen>
   <dimen name="faq_single_padding_bottom">132dp</dimen>
+  <dimen name="faq_single_activity_scroll_padding_Bottom">0dp</dimen>
```
Rename to `faq_single_activity_scroll_padding_bottom`

rishidyno replied:
This has also been renamed to `faq_single_activity_scroll_padding_bottom`  in dimens.xml sw600dp-land.

rt4914 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -409,6 +409,7 @@
   <dimen name="faq_single_padding_top">22dp</dimen>
   <dimen name="faq_single_padding_end">28dp</dimen>
   <dimen name="faq_single_padding_bottom">60dp</dimen>
+  <dimen name="faq_single_activity_scroll_padding_Bottom">80dp</dimen>
```
Rename to `faq_single_activity_scroll_padding_bottom`

rishidyno replied:
same in dimens.xml

rt4914 reviewed (changes requested):
@rishidyno You can notice in screenshots that the tablet-landscape screen has changed. Also looking at the code I would suggest you to merge only `layout` and `layout-land` file.
So in the end instead of three you will have only two files: `layout/faq_single_activity` and `layout-sw600dp-land/faq_single_activity`

rishidyno committed:
Renamed to faq_single_activity_scroll_padding_Bottom to faq_single_activity_scroll_padding_bottom

rishidyno committed:
Renamed to faq_single_activity_scroll_padding_Bottom to faq_single_activity_scroll_padding_bottom

rishidyno committed:
Renamed to faq_single_activity_scroll_padding_Bottom to faq_single_activity_scroll_padding_bottom

rishidyno committed:
Renamed to faq_single_activity_scroll_padding_Bottom to faq_single_activity_scroll_padding_bottom

rishidyno commented:
@rt4914 @anandwana001 @prayutsu  I have done the required changes. Please review it.

prayutsu commented on .idea/deploymentTargetDropDown.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="deploymentTargetDropDown">
+    <targetSelectedWithDropDown>
+      <Target>
+        <type value="QUICK_BOOT_TARGET" />
+        <deviceKey>
+          <Key>
+            <type value="VIRTUAL_DEVICE_PATH" />
+            <value value="C:\Users\RISHI RAJ\.android\avd\Pixel_C_API_30_1.avd" />
+          </Key>
+        </deviceKey>
+      </Target>
+    </targetSelectedWithDropDown>
+    <timeTargetWasSelectedWithDropDown value="2021-08-04T06:54:43.519797800Z" />
+  </component>
+</project>
```
Revert these changes

rishidyno replied:
Done

prayutsu reviewed (changes requested):
@rishidyno Left one comment, also when you add the suggested changes, then you should reply back in the conversation threads so that the reviewer is aware that you have made that particular change or you faced some problem.

prayutsu commented on app/src/main/res/values-sw600dp-land/booleans.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <bool name="faq_single_activity_scroll_fill_view_port">true</bool>
+</resources>
```
Remove this if we are not actually using this boolean anywhere.

rishidyno replied:
Ya! I have removed this now unnecessary file.

prayutsu commented on app/src/main/res/layout/faq_single_activity.xml:
```diff
@@ -40,10 +41,12 @@
         android:background="@color/oppiaGreyBackground"
         android:clipToPadding="false"
+        android:fillViewport="@bool/faq_single_activity_scroll_fill_view_port"
         android:overScrollMode="never"
```
We can probably remove this attribute now ( As we are keeping the tablet landscape layout)

rishidyno replied:
Done

prayutsu reviewed (changes requested):
Left some more comments, PTAL @rishidyno

rishidyno committed:
Delete deploymentTargetDropDown.xml

rishidyno committed:
deleted boolenas.xml sw600dp-land 

Now we are having 2 versions of faq_single_activity we don't need to mention fillViewport attribute so i have deleted this file

rishidyno committed:
deleted unnecessary fillviewPort attribute from activity

rishidyno committed:
Deleted 2 unnecessary attributes in default layout

rt4914 commented:
@rishidyno Please reply to the open comments so that reviewer can know what exactly you did for that particular comment.

rishidyno committed:
Delete deploymentTargetDropDown.xml

rishidyno commented:
@rt4914 All addressed changes are done.  I have retained all the attributes needed and deleted unnecessary ones.

rishidyno commented:
> @rishidyno Please reply to the open comments so that reviewer can know what exactly you did for that particular comment.

Sir! sorry for the inconvenience. I will take care of this in the future.

rishidyno committed:
Merge branch 'oppia:develop' into merge-faq_single_activity-into-single-layout-file

rishidyno commented:
@rt4914  Please review this PR

rishidyno committed:
Merge branch 'oppia:develop' into merge-faq_single_activity-into-single-layout-file

rishidyno committed:
Merge branch 'oppia:develop' into merge-faq_single_activity-into-single-layout-file

prayutsu reviewed (approved):
LGTM, thanks @rishidyno.

rishidyno commented:
Ptal @rt4914   

 All tests have passed and I have updated the project.

rishidyno committed:
Merge branch 'oppia:develop' into merge-faq_single_activity-into-single-layout-file

rishidyno committed:
Merge branch 'oppia:develop' into merge-faq_single_activity-into-single-layout-file

rt4914 reviewed (approved):
LGTM, thanks.

rishidyno commented:
> LGTM, thanks.

Thanks @rt4914
